src/os/bluestore/BlueFS.cc

```diff
@@ -2229,6 +2388,174 @@ int64_t BlueFS::_read_random(
   return ret;
 }
 
+void BlueFS::_wal_update_size(FileRef file, uint64_t flush_offset, uint64_t increment) {
+  using WALLength = File::WALFlush::WALLength;
+  using WALMarker = File::WALFlush::WALMarker;
+  
+  dout(20) 
+      << fmt::format(
+        "{} updating WAL file {} for range {:#x}~{:#x} limit is {:#x}", 
+        __func__, file->fnode.ino, flush_offset, increment, file->fnode.wal_limit) 
+      << dendl;
+  ceph_assert((flush_offset == 0 && file->wal_flushes.empty()) || (flush_offset == file->wal_flushes.back().end_offset()));
+
+  FileReader *h = new FileReader(file, cct->_conf->bluefs_max_prefetch, false, true);
+  
+  size_t header_size = File::WALFlush::header_size();
+
+  uint64_t flush_end = flush_offset + increment;
+  while (flush_offset < file->fnode.wal_limit) {
+    // read first part of wal flush
+    bufferlist bl;
+    bluefs_wal_header_t header;
+    
+    uint64_t read_result = (uint64_t)_read(h, flush_offset, header_size, &bl, nullptr);
+    if (read_result < header_size) {
+      dout(20) << fmt::format("{} cannot read wal header, most likely we are out of bounds. flush_offset={:#X}", __func__, flush_offset) << dendl;
+      break;
+    }
+    
+    dout(30) << __func__ << " result \n";
+    bl.hexdump(*_dout);
+    *_dout << dendl;
+    auto buffer_iterator = bl.cbegin();
+    decode(header, buffer_iterator);
```

Comment:
You can wrap decode() call with try..catch and break reading if decode fails. We've been successfully using the same technique in a couple other locations in BlueFS....

Reply:
So what are your thoughts about ?

Reply:
added it